src/main/java/racingcar/domain/CarMovement.java

```diff
@@ -0,0 +1,30 @@
+package racingcar.domain;
+
+import static racingcar.config.CarMovementConfig.*;
+
+public class CarMovement {
+	private final NumberGenerator numberGenerator;
+
+	public CarMovement(NumberGenerator numberGenerator) {
+		this.numberGenerator = numberGenerator;
+	}
+
+	public boolean isMoveForward() {
+		int number = numberGenerator.generateNumber();
+		if (isStop(number)) {
+			return false;
+		}
+		if (isForward(number)) {
+			return true;
+		}
+		return false;
+	}
```

Comment:
메서드 분리 💯
if문을 지나 마지막 리턴값은 어떤 경우에 발생하나요? 멈추거나 전진하는것 이외의 동작이 있나요?

Reply:
20라인 return false의 경우 움직이지 않습니다.

정지에 대한 숫자가 아니고
전진에 대한 숫자가 아니라면

"정지해서 아무것도 하지 말아라" 라는 것을 생각했었습니다.

하지만 소니가 말씀하신 이유가 
return false의 경우 어떤 동작이 일어날지 명확하지 않고
만약 false로 return 됐을 때 자동차가 후진하게 된다면 요구사항에 부합하게 되는거라는 생각을 할 수 있었습니다. ✌️

때문에 범위에 부합한 number가 들어왔을 경우는 애플리케이션 내 오류라고 판단하는게 맞다는 의견을 내고 싶습니다.

다음과 같이 코드를 변경하겠습니다. 😁
```suggestion

	public boolean isMoveForward() {
		int number = numberGenerator.generateNumber();
		if (isStop(number)) {
			return false;
		}
		if (isForward(number)) {
			return true;
		}
		throw new IllegalStateException(CAR_MOVEMENT_NUMBER_EXCEPTION.getMessage());
	}
```

Reply:
현재 코드상 0~9 이외의 값이 들어올 수 있나요?

Reply:
현재 코드상에서 0-9 이외의 값이 들어올 수 없습니다.
랜덤으로 나올 수 있는 범위를 0-9로 지정해놨기 때문입니다.

하지만 랜덤의 범위가 0-10으로 바뀌는 순간 애플리케이션이 오작동 할 수 있다고 판단하여
이외의 값이 들어올 경우에 예외를 던져서 인지 하려고 했습니다.
현재 요구사항은 정지, 전진 두 가지 이동 방법밖에 없기 때문에 아래 코드와 같이 전진이 아니면 return false로 정지만 시켜도 될 거 같습니다.
```java
public boolean isMoveForward() {
   if (isForward(number) {
    return true;
  }
  return false;
}
```

하지만 저는 만약에 이동 방식이 "정지", "전진"이 아닌 "후진", "전진" 이었다면
기존 코드에서 마지막에 return false 할 경우에는 후진하여 오작동 할 수 있습니다.
```java
public boolean isMoveForward() {
  int number = numberGenerator.generateNumber();
  if (isStop(number)) {
    return false;
  }
  if (isForward(number)) {
    return true;
  }
  return false;
}
```

그러한 요구상이 바뀌었을 때 테스트 코드에서 실패하여 고칠 수 있겠지만
모든 if 조건에 부합 했을 경우는 없어야 한다는 정보를 주고 싶었습니다.
때문에 아래와 같은 코드로 변경하게 되었습니다.
```java
public boolean isMoveForward() {
  int number = numberGenerator.generateNumber();
  if (isStop(number)) {
    return false;
  }
  if (isForward(number)) {
    return true;
  }
  throw new IllegalStateException(CAR_MOVEMENT_NUMBER_EXCEPTION.getMessage());
}
```